src/Controller.js

```diff
@@ -0,0 +1,123 @@
+import FunctionTree from 'function-tree'
+import Module from './Module'
+import DefaultModel from './DefaultModel'
+import {ensurePath, isDeveloping, throwError} from './utils'
+import VerifyInputProvider from './providers/VerifyInput'
+import ContextProvider from 'function-tree/providers/Context'
+import StateProvider from './providers/State'
+import DebuggerProvider from './providers/Debugger'
+import ControllerProvider from './providers/Controller'
+import {EventEmitter} from 'events'
+import {dependencyStore as computedDependencyStore} from './Computed'
+
+/*
+  The controller is where everything is attached. The devtools
+  and router is attached directly. Also a top level module is created.
+  The controller creates the function tree that will run all signals,
+  based on top level providers and providers defined in modules
+*/
+class Controller extends EventEmitter {
+  constructor({state = {}, routes = {}, signals = {}, providers = [], modules = {}, router, devtools}) {
+    super()
+    this.flush = this.flush.bind(this)
+    this.devtools = devtools
+    this.model = new DefaultModel({})
```

Comment:
Ah, yes, I have been thinking about this. With devtools I am thinking:

```js
Controller({
  devtools: Devtools({freezeState: true})
})
```

Unsure if it should be default though. Maybe it should be. And if you do not care about timeTravel and this freeze thing, you do:

```js
Controller({
  devtools: Devtools({freezeState: false, storeMutations: false})
})
```

I will officially put it on my list :)